Content.Server/Cloning/CloningConsoleSystem.cs

```diff
@@ -0,0 +1,312 @@
+        {
+            cloneConsoleComp.CloningPod = null;
+            cloneConsoleComp.GeneticScanner = null;
+            if (TryComp<TransformComponent>(Owner, out var transformComp) && transformComp.Anchored)
+            {
+                var grid = _mapManager.GetGrid(transformComp.GridID);
+                var coords = transformComp.Coordinates;
+                foreach (var entity in grid.GetCardinalNeighborCells(coords))
+                {
+                    if (TryComp<MedicalScannerComponent>(entity, out var geneticScanner) && geneticScanner.ConnectedConsole == null)
+                    {
+                        cloneConsoleComp.GeneticScanner = entity;
+                        geneticScanner.ConnectedConsole = Owner;
+                        continue;
+                    }
+
+                    if (TryComp<CloningPodComponent>(entity, out var cloningPod) && cloningPod.ConnectedConsole == null)
+                    {
+                        cloneConsoleComp.CloningPod = entity;
+                        cloningPod.ConnectedConsole = Owner;
+                    }
+                }
+            }
+            UpdateUserInterface(cloneConsoleComp);
+            return;
+        }
+
+        public void TryEject(EntityUid uid, CloningConsoleComponent consoleComponent)
+        {
+            if (consoleComponent.CloningPod == null)
+                return;
+
+            if (!TryComp<CloningPodComponent>(consoleComponent.CloningPod, out var cloningPod))
+            {
+                consoleComponent.CloningPod = null;
+                return;
+            }
+
+            _cloningSystem.Eject(consoleComponent.CloningPod.Value, cloningPod);
+        }
+
+        public void TryClone(EntityUid uid, CloningConsoleComponent consoleComponent)
+        {
+             if (consoleComponent.GeneticScanner == null || consoleComponent.CloningPod == null)
+                return;
+
+            if (!TryComp<CloningPodComponent>(consoleComponent.CloningPod, out var cloningPod))
+            {
+                consoleComponent.CloningPod = null;
+                return;
+            }
+
+             if (!TryComp<MedicalScannerComponent>(consoleComponent.GeneticScanner, out var scanner))
+            {
+                consoleComponent.GeneticScanner = null;
+                return;
+            }
+
+            if (scanner.BodyContainer.ContainedEntity is null)
+                return;
+
+            if (!TryComp<MindComponent>(scanner.BodyContainer.ContainedEntity.Value, out var mindComp) || mindComp.Mind == null)
+                return;
+
+            var mind = mindComp.Mind!;
+            var mindUser = mind.UserId;
+            if (mindUser.HasValue == false || mind.Session == null)
+                return;
+            var profile = GetPlayerProfileAsync(mindUser.Value);
+            bool cloningSuccessful = _cloningSystem.TryCloning(consoleComponent.CloningPod.Value, mind, profile, cloningPod);
+        }
```

Comment:
You should just make these cloning system partials given they're all co-dependent anyway.